src/schemes/post.ts

```diff
@@ -4,8 +4,8 @@ const postSchema = Joi.object({
   image: Joi.string()
     .regex(/(https?:\/\/.*\.(?:png|jpg|jpeg|gif))/i)
     .allow(''),
-  AnimalId: Joi.number().required(),
-  TagId: Joi.number().required(),
+  Animal: { id: Joi.number().required(), name: Joi.string().required() },
```

Comment:
post controllers were designed to take an id only according to the issues instructing how to build them. this is somewhat considered a redesign to that logic that should've been documented by an issue or a comment on this pr at least